src/CommonITILObject.php

```diff
@@ -4257,18 +4249,22 @@ public static function processMassiveActionsForOneItemtype(
                 $input = $ma->getInput();
                 foreach ($ids as $id) {
                     if ($item->can($id, UPDATE)) {
-                        $linkclass = new $item->userlinkclass();
-                        foreach ($linkclass->getActors($id) as $users) {
-                            foreach ($users as $data) {
-                                $data['use_notification'] = $input['use_notification'];
-                                $linkclass->update($data);
+                        if (is_a($item->userlinkclass, CommonITILActor::class, true)) {
+                            $linkclass = new $item->userlinkclass();
```

Comment:
The method is protected, but I think it would work as you call it from the same class.

```suggestion
                            $linkclass = $item->getActorObjectForItem(User::class);
```